src/framework/components/rigid-body/system.js

```diff
@@ -522,38 +529,16 @@ Object.assign(pc, function () {
             return r && (r.hasEvent("collisionstart") || r.hasEvent("collisionend") || r.hasEvent("contact"));
         },
 
-        onUpdate: function (dt) {
-            // #ifdef PROFILER
-            this._stats.physicsStart = pc.now();
-            // #endif
-
-            // Check to see whether we need to update gravity on the dynamics world
-            var gravity = this.dynamicsWorld.getGravity();
-            if (gravity.x() !== this.gravity.x || gravity.y() !== this.gravity.y || gravity.z() !== this.gravity.z) {
-                gravity.setValue(this.gravity.x, this.gravity.y, this.gravity.z);
-                this.dynamicsWorld.setGravity(gravity);
-            }
-
-            // Update the transforms of all bodies
-            this.dynamicsWorld.stepSimulation(dt, this.maxSubSteps, this.fixedTimeStep);
-
-            // Update the transforms of all entities referencing a body
-            var components = this.store;
-            for (var id in components) {
-                if (components.hasOwnProperty(id)) {
-                    var entity = components[id].entity;
-                    var componentData = components[id].data;
-                    if (componentData.body && componentData.body.isActive() && componentData.enabled && entity.enabled) {
-                        if (componentData.type === pc.BODYTYPE_DYNAMIC) {
-                            entity.rigidbody.syncBodyToEntity();
-                        } else if (componentData.type === pc.BODYTYPE_KINEMATIC) {
-                            entity.rigidbody._updateKinematic(dt);
-                        }
-                    }
-
-                }
-            }
-
+        /**
+         * @private
+         * @function
+         * @name pc.RigidBodyComponentSystem#_checkForCollisions
+         * @description Checks for collisions and fires collision events
+         * @param {pc.Entity} dynamicsWorld - The pointer to the dynamics world that invoked this callback.
+         * @param {pc.Entity} timeStep - The amount of simulation time processed in the last simulation tick.
+         * @returns {void}
+         */
+        _checkForCollisions: function (dynamicsWorld, timeStep) {
             // Check for collisions and fire callbacks
             var dispatcher = this.dynamicsWorld.getDispatcher();
```

Comment:
I'd change `this.dynamicsWorld` to just `dynamicsWorld` since it's a parameter of the function now.

Reply:
So because of the way WASM handles that callback, the dyanmicsWorld is returned at an integer pointer and you would have to make an extra call every step to get the underlying data which seems like it would have an impact on performance. They could be removed from the function signature and it will still work fine, but I figured in the future if you wanted to keep track of exact simulation time you would need the timeStep and would have to define the dynamicsWorld to get that.

Reply:
Gotcha. OK, then let's leave this 'as is'.